extension/js/common/core/common.ts

```diff
@@ -315,6 +315,11 @@ export class Url {
     return `${urlParts[0]}?${params.toString()}`;
   };
 
+  public static replaceUrlParam = (url: string, key: string, value: string) => {
+    const regex = new RegExp(`(${key}=).*?(&|$)`);
```

Comment:
This also reacts to prefixed keys, like `xyzacctEmail=`.
We can specify a non-capturing group requiring the match to start from either ? or & (or beginning of string?)
`((?:\?|&)acctEmail=).*?(&|$)`
I'd also suggest to add case-insensitive option 'i'.

Reply:
Perhaps, for better security we can leave the regex as it is to scrub even prefixed parameters, but adding case-insensitivity option 'i' for better security would also be good.